### PR TITLE
Add the ability to delete guides.

### LINKED ISSUE #457: Delete Guides
Author should be able to delete their Guides

### PR DESCRIPTION
**Pull request by guitsaru**
resolves #457

### CODE REVIEW

File: app/controllers/guides_controller.rb [hunk 1]
```diff
@@ -21,4 +21,12 @@ def edit
       redirect_to @guide
     end
   end
+
+  def destroy
+    @guide = current_user.guides.find(params[:id])
```
Comment (simonv3) [inline app/controllers/guides_controller.rb]: You could do this through the mutation as well, that way we keep all of the code to do with the actual destruction in one place.

File: spec/controllers/api/api_guides_controller_spec.rb [hunk 2]
```diff
@@ -74,6 +74,33 @@
...
+    end
+
+    it 'only destroys guides owned by the user' do
+      sign_in user
+      delete :destroy, id: FactoryGirl.create(:guide)
+      expect(json['error']).to include(
+        'can only destroy guides that belong to you.')
+      expect(response.status).to eq(401)
+    end
+  end
```
Comment (simonv3) [inline spec/controllers/api/api_guides_controller_spec.rb]: Thanks for writing these!

**Comment by simonv3** [comment]
Thanks for the PR! This looks really good. Can you have a look at my comment about rolling the `guides_controller` code into the mutation as well and let me know your thoughts on that?

Cheers,
Simon

File: app/controllers/guides_controller.rb (hunk 1 above)
Comment (guitsaru) [inline app/controllers/guides_controller.rb]: Yep, thanks for pointing this out.

**Pull request merged**